PIL/Image.py

```diff
@@ -583,21 +583,27 @@ def _copy(self):
 
     def _dump(self, file=None, format=None, **options):
```

Comment:
This wouldn't resolve the type-related issue, which is that `file` is effectively an `Optional[Text]` (`Union[None, Text]`).

Assigning to it isn't treated (by mypy) as removing the `None`-ness, so use of the variable later assumes it still may be `None`. Alternatively the code can be enclosed within a conditional checking for `None`-ness, but here that would probably be a duplicating of the subsequent code since format would need to be checked in each case.